Neos.ContentGraph.DoctrineDbalAdapter/src/ContentGraphFactory.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Neos\ContentGraph\DoctrineDbalAdapter;
+
+use Neos\ContentGraph\DoctrineDbalAdapter\Domain\Repository\ContentGraph;
+use Neos\ContentGraph\DoctrineDbalAdapter\Domain\Repository\NodeFactory;
+use Neos\ContentRepository\Core\ContentGraphFactoryInterface;
+use Neos\ContentRepository\Core\Infrastructure\DbalClientInterface;
+use Neos\ContentRepository\Core\NodeType\NodeTypeManager;
+use Neos\ContentRepository\Core\Projection\Workspace\Workspace;
+use Neos\ContentRepository\Core\Projection\Workspace\WorkspaceStatus;
+use Neos\ContentRepository\Core\SharedModel\ContentRepository\ContentRepositoryId;
+use Neos\ContentRepository\Core\SharedModel\Exception\ContentStreamDoesNotExistYet;
+use Neos\ContentRepository\Core\SharedModel\Workspace\ContentStreamId;
+use Neos\ContentRepository\Core\SharedModel\Workspace\WorkspaceDescription;
+use Neos\ContentRepository\Core\SharedModel\Workspace\WorkspaceName;
+use Neos\ContentRepository\Core\SharedModel\Workspace\WorkspaceTitle;
+
+/**
+ * @internal only used inside the
+ * @see ContentGraphFinder
+ */
+readonly final class ContentGraphFactory implements ContentGraphFactoryInterface
+{
+    public function __construct(
+        private DbalClientInterface $client,
+        private NodeFactory $nodeFactory,
+        private ContentRepositoryId $contentRepositoryId,
+        private NodeTypeManager $nodeTypeManager,
+        private string $tableNamePrefix
+    ) {
+    }
+
+    public function buildForWorkspace(WorkspaceName $workspaceName): ContentGraph
+    {
+        // FIXME: Should be part of this projection, this is forbidden
+        $tableName = strtolower(sprintf(
+            'cr_%s_p_%s',
+            $this->contentRepositoryId->value,
+            'Workspace'
```

Comment:
Nitpick, but to save one needless character conversion:
```suggestion
            'workspace'
```
or even just use
```php
$tableName = sprintf('cr_%s_p_workspace', $this->contentRepositoryId->value);
```
like done in other places? The CR ID can only be lower case

Reply:
Yep, this was literally a last minute thing because the tests ran for me (case insensitive file system for the db table files -> need to move them) vs. case sensitive on GH actions. Didn't check if CR ID is lowercase but yeah lets fix it this way. it would become nicer when the workspace projection is part of the CG adapter anywys as then we would do the contentGraphTableNames->workspace() thing 😄